backend/inspirehep/records/serializers/bibtex.py

```diff
@@ -16,6 +19,30 @@
 
 
 class BibtexWriter(Writer):
+    def _encode(self, text):
+        import latexcodec  # NOQA
+
+        if "$" not in text:
+            return codecs.encode(text, "ulatex+{}".format(self.encoding))
+        else:
+            math_expression_regex = re.compile(r"(?<!\\)\$.*?\$")
+            found_math_expressions = [
+                f"{expression}"
+                for expression in re.findall(math_expression_regex, text)
+            ]
+            encoded_text = codecs.encode(text, "ulatex+{}".format(self.encoding))
+            encoded_math_expressions = re.findall(math_expression_regex, encoded_text)
+            for encoded_math_expression, found_math_expression in zip(
+                encoded_math_expressions, found_math_expressions
+            ):
+                encoded_text = encoded_text.replace(
+                    encoded_math_expression, found_math_expression
+                )
+            assert len(found_math_expressions) == len(
+                encoded_math_expressions
+            ), f"number of math parts in LaTeX-encoded string is different for '{text}'."
```

Comment:
in general it's better to put assertions doing sanity checks before the code that uses the assumption, so
1. you don't do useless work in case it fails
2. you can rely on the assumption being true when you need it